### ISSUE TITLE
construct_tdigest crashes if the array sizes and count dont match

### ISSUE THREAD

**Issue by kaikalur** [prestodb/presto#22146]
Fix construct_tdigest to explicitly check for count/centroid mean/weight arrays size mismatch (or fix it if it can be different). Looks like there is some assumption that they are all same size. simple example:

```
presto:oculus> select construct_tdigest(array[1.],array[1.,2],1.,1.,1.,1.,10);
Query 20240310_162224_36022_xr3hz failed: end index (80) must not be greater than size (16)
```

**Comment by jameswang2015** [comment]
@kaikalur I can understand the size of centroid.mean and weight should be always the same, but the count doesn't have to be same, since the count should be the number of the original data points. 

cc: Daniel Izcovich

**Comment by kaikalur** [comment]
yeah I thought so too but could't confirm it. Please verify and feel free to fix it to exit gracefully upon invalid arguments

**Comment by tdcmeehan** [comment]
IIRC the count represents the number of entries per centroid, so you'd need the count to be consistent as well.
